Entity/Transaction.php

```diff
@@ -459,17 +459,17 @@ public function getSuccess()
      */
     public function setMessages(ArrayCollection $messages = null)
     {
-        $this->messages = $messages->isEmpty() ? null : $messages;
+        $this->messages = empty($messages) || $messages->isEmpty() ? null : $messages->toArray();
```

Comment:
What are the cases when we need to check for empty? We expect it to be ArrayCollection or null. If empty collection was passed, it should stay empty collection, and not be converted. Probably its better to create separate method to check if messages exists, and then call getMessages.